topic-operator/src/main/java/io/strimzi/operator/topic/TopicSerialization.java

```diff
@@ -45,52 +48,47 @@
     public static final String JSON_KEY_REPLICAS = "replicas";
     public static final String JSON_KEY_CONFIG = "config";
 
+    private static void validateConfiguration(Reconciliation reconciliation, KafkaTopic kafkaTopic, String desiredVersion, KafkaTopicConfiguration configuration) {
+        List<String> errorsInConfig = configuration.validate(desiredVersion);
+
+        if (!errorsInConfig.isEmpty()) {
+            for (String error : errorsInConfig) {
+                LOGGER.warnCr(reconciliation, "KafkaTopic {}/{} has invalid spec.config: {}",
+                        kafkaTopic.getMetadata().getNamespace(),
+                        kafkaTopic.getMetadata().getName(),
+                        error);
```

Comment:
Is there much value in logging, when we're throwing the exception. Won't the catcher of the exception both log the message and also set the CR condition to include the message? Not logging here would avoid needing to pass the reconciliation so deep.